src/box/memtx_rtree.cc

```diff
@@ -332,9 +507,10 @@ memtx_rtree_index_create_iterator(struct index *base, enum iterator_type type,
 	struct memtx_rtree_index *index = (struct memtx_rtree_index *)base;
 	struct memtx_engine *memtx = (struct memtx_engine *)base->engine;
 
-	if (pos != NULL) {
-		diag_set(UnsupportedIndexFeature, base->def, "pagination");
-		return NULL;
+	if (pos != NULL && type != ITER_NEIGHBOR) {
+		diag_set(UnsupportedIndexFeature, base->def,
+			 "pagination with wrong iterator type. "
+			 "Only for neighbor iterator is supported");
```

Comment:
You don't return `NULL` here, hence, the error is not thrown.
Please, fix, and cover with a test.